src/runtime/audience-action-flow-test.js

```diff
@@ -131,14 +134,76 @@ describes.realWin('AudienceActionFlow', {}, (env) => {
         .withExactArgs(Constants.USER_TOKEN, userToken, true)
         .exactly(timesStoreSetCalled);
 
+      let toast;
+      const toastOpenStub = sandbox
+        .stub(Toast.prototype, 'open')
+        .callsFake(function () {
+          toast = this;
+        });
+
+      await audienceActionFlow.start();
+      const completeAudienceActionResponse =
+        new CompleteAudienceActionResponse();
+      completeAudienceActionResponse.setActionCompleted(true);
+      completeAudienceActionResponse.setSwgUserToken(userToken);
+      const messageCallback =
+        messageMap[completeAudienceActionResponse.label()];
+      messageCallback(completeAudienceActionResponse);
+
+      entitlementsManagerMock.verify();
+      storageMock.verify();
+      expect(toastOpenStub).to.be.called;
+      expect(toast).not.to.be.null;
+      expect(toast.src_).to.contain('flavor=basic');
+    });
+  });
+
+  [
+    {
+      action: 'TYPE_REGISTRATION_WALL',
+      toastMessage: 'You have registered before.',
+    },
+    {
+      action: 'TYPE_NEWSLETTER_SIGNUP',
+      toastMessage: 'You have signed up before.',
+    },
+  ].forEach(({action, toastMessage}) => {
+    it('correctly handles a CompleteAudienceActionResponse with action not completed when returned from the flow', async () => {
```

Comment:
same here, "correctly" seems redundant. test name can probably be improved.

Reply:
The description is updated.